packages/transition-common/src/services/nodes/__tests__/NodeGeographyUtils.test.ts

```diff
@@ -0,0 +1,86 @@
+import { v4 as uuidV4 } from 'uuid';
+import { EventEmitter } from 'events';
+import { distance as turfDistance, point as turfPoint } from '@turf/turf';
+
+import * as Status from 'chaire-lib-common/lib/utils/Status';
+
+import Node, { NodeAttributes } from '../Node';
+import { proposeNames } from '../NodeGeographyUtils';
+
+const mockSocket = new EventEmitter();
+mockSocket.on('osm.streetsAroundPoint', (nodeGeojson, radiusAroundMeters, callback) => {
+    if (nodeGeojson.properties.code === '123') { // has intersection
+        callback(Status.createOk([
+            {
+                type: 'Feature',
+                properties: { name: 'Street 1' },
+                geometry: { type: 'LineString', coordinates: [[-73.9980, 45.0000], [-73.9970, 45.0010]] },
+            },
+            {
+                type: 'Feature',
+                properties: { name: 'Street 2' },
+                geometry: { type: 'LineString', coordinates: [[-73.9970, 45.0010], [-73.9960, 45.0020]] },
+            },
+        ] as GeoJSON.Feature[]));
+    } else if (nodeGeojson.properties.code === '234') { // no intersection
+        callback(Status.createOk([
+            {
+                type: 'Feature',
+                properties: { name: 'Street 3' },
+                geometry: { type: 'LineString', coordinates: [[-73.9980, 45.0000], [-73.9965, 45.0015]] },
+            },
+            {
+                type: 'Feature',
+                properties: { name: 'Street 4' },
+                geometry: { type: 'LineString', coordinates: [[-73.9965, 45.0000], [-73.9950, 45.0015]] },
+            },
+        ] as GeoJSON.Feature[]));
+    } else {
+        callback({ status: 'error', error: 'Invalid radius' });
+    }
+});
+
+describe('proposeNames', () => {
+    test('should return intersection names within the specified radius, ordered by distance', async () => {
+        const node = new Node({
+            id: uuidV4(),
+            code: '123',
+            geography: { type: 'Point' as const, coordinates: [-73.9975, 45.0005] },
+        }, true);
+        const maxRadiusMeters = 100;
+
+        const result = await proposeNames(mockSocket, node, maxRadiusMeters);
+
+        const distance1 = turfDistance(turfPoint([-73.9975, 45.0005]), turfPoint([-73.9970, 45.0010]));
```

Comment:
Ideally, instead of hardcoding the values here, you could have put the coordinates in variables in the test and use them here and in the features above. Now if you change one, you'll need to update all the tests and figure out why they fail, if they fail, if they don't just silently do nothing...

Reply:
coordinates of points are different, but I added variables for streets